ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh

```diff
@@ -60,15 +63,31 @@ MAX_ZONES_COUNT="${#ZONES[@]}"
 # Save max zones count information to ${SHARED_DIR} for use in other scenarios
 echo "${MAX_ZONES_COUNT}" >> "${SHARED_DIR}/maxzonescount"
 
+existing_zones_setting=$(/tmp/yq r "${CONFIG}" 'controlPlane.platform.aws.zones')
 
-ZONES_COUNT=${ZONES_COUNT:-2}
-ZONES=("${ZONES[@]:0:${ZONES_COUNT}}")
-ZONES_STR="[ "
-ZONES_STR+=$(join_by , "${ZONES[@]}")
-ZONES_STR+=" ]"
-echo "AWS region: ${REGION} (zones: ${ZONES_STR})"
+if [[ ${existing_zones_setting} == "" ]]; then
+  ZONES_COUNT=${ZONES_COUNT:-2}
+  ZONES=("${ZONES[@]:0:${ZONES_COUNT}}")
+  ZONES_STR="[ $(join_by , "${ZONES[@]}") ]"
+  echo "AWS region: ${REGION} (zones: ${ZONES_STR})"
+  PATCH="${SHARED_DIR}/install-config-zones.yaml.patch"
+  cat > "${PATCH}" << EOF
+controlPlane:
+  platform:
+    aws:
+      zones: ${ZONES_STR}
+compute:
+- platform:
+    aws:
+      zones: ${ZONES_STR}
+EOF
+  /tmp/yq m -x -i "${CONFIG}" "${PATCH}"
+else
+  echo "zones already set in install-config.yaml, skipped"
+fi
 
-cat >> "${CONFIG}" << EOF
+PATCH="${SHARED_DIR}/install-config-common.yaml.patch"
```

Comment:
nit: I think we want `PATCH` to live in `/tmp`, see #20865 where I try to fix up our existing steps.  I'm fine rebasing my PR if yours lands first with the `SHARED_DIR` here, though.

Reply:
Thanks. I thought I can find these patch file in job's artifacts for trouble shooting, but sound like I did not find out them. So I am okay with move these patch files to `/tmp`. 

Yeah, I prefer to land this PR firstly, then your rebase them. Thanks in advance.